### ISSUE TITLE
Generalize IndexFeatureFile to IndexFile by adding support for indexing bams and fastas

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#4349]

**Comment by chandrans** [comment]
Hey David. This will be a tool again in the next release? Users have been asking about generating VCF index, and I have pointed them to IGV or tabix or samtools. I think this has been disabled in the latest point release.

**Comment by droazen** [comment]
@chandrans `IndexFeatureFile` has been enabled and runnable in all of the 4.x releases so far. It was only disabled temporarily in one of the betas.

**Comment by droazen** [comment]
(It's been re-enabled since August 2017, by the way! This is the PR that re-enabled it: https://github.com/broadinstitute/gatk/pull/3425)

**Comment by chandrans** [comment]
Yes, you are right. Sorry, I must have tested with a beta version and thought it was disabled. It does work in 4.0.1.1 :) I will have to let users know.
